src/module.js

```diff
@@ -34,74 +37,121 @@ const execute = (command, log) => {
     });
 };
 
+function escapeRegExp (string) {
+    // $& means the whole matched string
+    return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+}
+
+function uuidFromVMName (vmName, uuid, log) {
+    return new Promise((resolve, reject) => {
+        if (uuid) {
+            return resolve(uuid);
+        }
+
+        log.info(`Running 'VBoxManage list vms' to locate installed VM named '${ vmName }'`);
+        execute('VBoxManage list vms', log)
+            .then((result) => {
+                const escapedVmName = escapeRegExp(vmName);
+                // eslint-disable-next-line no-useless-escape
+                const regex = new RegExp(escapedVmName + '.*\{(.+?)\}', 'igm');
+                const m = regex.exec(result);
+
+                if (m && m.length === 2) {
+                    const parsedUuid = m[1];
+
+                    log.info(`Located VM UUID '${ parsedUuid }'`);
+
+                    return resolve(parsedUuid);
+                }
+
+                return reject(new Error(`No virtual machine installed named '${ vmName }'`));
+            })
+            .catch((err) => {
+                return reject(err);
+            });
+    });
+}
+
 function VirtualBoxIE11Browser (args, baseBrowserDecorator, logger) {
     baseBrowserDecorator(this);
 
     const kill = !args.keepAlive;
     const log = logger.create('launcher');
     const snapshot = args.snapshot;
-    const uuid = args.uuid;
 
     // Just override this method to prevent the inherited one to be executed.
     this._start = () => {};
 
-    if (!uuid) {
-        log.error('Please specify the UUID of the virtual machine to use.');
+    if (!args.uuid && !args.vmName) {
+        log.error('Please specify the UUID or name of the virtual machine to use.');
 
         return;
     }
 
     this
         .on('kill', (done) => {
             if (kill) {
-                return execute(`VBoxManage controlvm {${ uuid }} acpipowerbutton`, log)
-                    .then(() => done())
-                    .catch(() => {
-                        log.error('Failed to turn of the vitual machine.');
-
-                        done();
+                uuidFromVMName(args.vmName, args.uuid, log)
+                    .then((uuid) => {
+                        return execute(`VBoxManage controlvm {${ uuid }} acpipowerbutton`, log)
```

Comment:
I left the original implementation of .on('kill') untouched, except for wrapping it inside the call to `uuidFromVMName`. The only other differences here should be just whitespace.